src/main/resources/lib/ssb/parts/upcomingReleases.ts

```diff
@@ -0,0 +1,230 @@
+      range: {
+        field: 'data.nextRelease',
+        from: 'dateTime',
+        gte: serverTime,
+        lte: endDate,
+      },
+    } as unknown as string,
+    filters: {
+      boolean: {
+        must: [
+          {
+            hasValue: {
+              field: 'language',
+              values: language === 'nb' ? ['nb', 'nn'] : ['en'],
+            },
+          },
+        ],
+        should: [
+          {
+            boolean: {
+              must: [
+                {
+                  hasValue: {
+                    field: 'data.articleType',
+                    values: ['statistics'],
+                  },
+                },
+                {
+                  hasValue: {
+                    field: 'data.status',
+                    values: ['A'],
+                  },
+                },
+                {
+                  exists: {
+                    field: 'data.statisticContentId',
+                  },
+                },
+              ],
+            },
+          },
+          // contentReleases
+          {
+            boolean: {
+              must: [
+                {
+                  hasValue: {
+                    field: 'type',
+                    values: [`${app.name}:upcomingRelease`],
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      },
+    },
+  })
+
+  const contents: (Content<UpcomingRelease, object> | ContentLight<Release>)[] = results.hits.map((hit) =>
+    connect(hit).get<Content<UpcomingRelease, object> | ContentLight<Release>>(hit.id)
+  )
+
+  const upcomingReleases: Array<PreparedStatistics> = contents.map((content) =>
+    isContentUpcomingRelease(content)
+      ? prepContentUpcomingRelease(content as Content<UpcomingRelease, object>, language, allMainSubjects)
+      : prepStatisticUpcomingRelease(content as ContentLight<Release>, language)
+  )
+
+  const filteredUpcomingReleasesStatistics = contents
+    .map((content) => filterUpcomingReleasesStatistics(content as ContentLight<Release>, language, serverTime, endDate))
+    .reduce((acc, curr) => {
+      if (curr.length) return acc.concat(curr)
+      else return acc
+    }, [])
+
+  const mergedStatisticsAndUpcomingStatisticsReleases = [
+    ...upcomingReleases,
+    ...(filteredUpcomingReleasesStatistics as Array<PreparedStatistics>),
+  ].sort((a, b) => {
+    return new Date(a.date as string).getTime() - new Date(b.date as string).getTime()
+  })
+
+  return {
+    total: mergedStatisticsAndUpcomingStatisticsReleases.length,
+    upcomingReleases: mergedStatisticsAndUpcomingStatisticsReleases,
+  }
+}
+
+function isContentUpcomingRelease(content: unknown) {
+  return (content as Content).type === `${app.name}:upcomingRelease`
+}
+
+function prepContentUpcomingRelease(
+  content: Content<UpcomingRelease, object>,
+  language: string,
+  allMainSubjects: Array<SubjectItem>
+): PreparedStatistics {
+  const date: string = content.data.nextRelease
```

Comment:
should this check if nextRelease exists, and if not get the field from content.data.date if that is present instead?